HotBackup documentation for 3.5 (#125)
arangodb/docs: 3.5/programs-arangobackup-limitations.md

transactions could prevent that from ever happening during a timeout period.
The same holds true for clusters, where this lock must now be obtained on all
database servers at the same time.

Especially in the cluster the result of these successively longer tries to
obtain the global transaction lock might become visible in periods of apparent
dead time. Locks might be obtained on some machines and and not on others, so
that the process has to be retried over and over. Every unsuccessful try would
then lead to the release of all partial locks.

ArangoSearch not Supported Yet
------------------------------

ArangoSearch views are not backed up and thus not restored yet.
Therefore, views have to be dropped and recreated after a restore.
This happens automatically in the background, but in particular in the
presence of large amounts of data, the recreation of the ArangoSearch
indexes can take some time after the restore. It is planned to rectify
this limitation in one of the next releases.

Note furthermore that a running query with views can prevent a restore
operation from happening whilst the query is running.

Windows not Supported Yet
-------------------------

> The first release of the hot backup feature (3.5.1) does not support

Reviewer: ```suggestion
The hot backup feature is not supported in the Windows version of ArangoDB
```

> the Windows version of ArangoDB. 

Reviewer: Is it planned for 3.5.2? If no, then we should avoid such a statement and just say that it's not supported yet.

Reviewer: Changed to not mention specific versions

> the Windows version of ArangoDB. 

Reviewer: ```suggestion
at this point in time.
```

Services on Single Server
-------------------------

> On a single server the installed services are not backed up and are

Reviewer: ```suggestion
On a single server the installed Foxx microservices are not backed up and are
```

therefore also not restored. This is because in single server mode
the service installation is done locally in the file system and does not
track the information in the `_apps` collection.

> In a cluster, the coordinators will after a restore eventually restore

Reviewer: ```suggestion
In a cluster, the coordinators will eventually restore
```

> the state of the services from the `_apps` collection.

Reviewer: ```suggestion
the state of the services from the `_apps` collection after a backup is restore.
```

Encryption at Rest
------------------

Currently, the hot backup simply takes a snapshot of the database files.
If one is using encryption at rest, then the backed up files will be
encrypted, with the encryption key that was used in the
instance which created the backup.

Such an encrypted backup can only be restored to an instance using the
same encryption key.

Replication and Hot Backup
--------------------------

Hot backups are not automatically replicated between instances. This is

> true both for the active/failover setup with 2 (or more) single servers,

Reviewer: ```suggestion
true for both the active failover setup with 2 (or more) single servers
```

> and for the data center to data center replication between clusters.

Reviewer: ```suggestion
and for the datacenter to datacenter replication between clusters.
```

> Simply take hot backups on all instances.

Reviewer: Is there a chance of running into trouble if the passive side is behind the active side when both are restored?
